### PR TITLE
Update github templates

### LINKED ISSUE #1732: Upgrade Github templates
## 1. General summary of the issue

Request to split current issue template on github repo into different templates for e.g. bugs and feature requests.

## 2. Describe Your Environment

Web browser

## 3. Expected Behavior

Dedicated templates for the common scenarios which is shown as a menu when creating a new issue. [Example](https://github.com/PowerShell/PowerShell/issues/new/choose)

- Bug Report
- Feature Request
- General question/support (provide template or just refer to https://github.com/pester/Pester#questions, the discussion issue etc.?)

- Link to slack channel for "community support"?

Maybe also update Pull Request template with Summary, Context and a checklist (remembered meaning title, tests required/added, breaking change yes/no) ?

## 4.Current Behavior

One issue template for all scenarios with a lot of comments 

## 5. Possible Solution

Create new [issue templates suited for different scenarios](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/configuring-issue-templates-for-your-repository), using the current template as a baseline for content.

## 6. Context

Lots of boilerplate that's not always necessary, but can feel like a "wall of text" for new user.
Simplify issue documentation by providing different templates for e.g. bugs and feature requests. Can include more specific steps and requirements for each scenario.

### PR DESCRIPTION
**Pull request by fflaten**
## 1. General summary of the pull request
PR suggest changes to the github templates used for the Pester repo. In replace of on large issue template, a use will get the choice between one per issue type:
- Bug report
- Feature request/idea
- Documentation issue
- General/blank issue for anything else

A link to the website for community support is also added

The text is also updated with new documentation-links, tips and updated script for collecting environment data (autodetects using currently loaded `Invoke-Pester`cmdlet).

The pull request template is also updated with new links and a checklist for common steps.

Fix #1732

### CODE REVIEW

**Comment by fflaten** [comment]
See demos for [issue templates](https://github.com/fflaten/githubsandbox/issues/new/choose) and [pull request template](https://github.com/fflaten/githubsandbox/compare/test?expand=1)

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,80 +1,38 @@
...
+**Environment data**
+
+<!--Please provide the output of a code provided below.
 
 Operating System, Pester version, and PowerShell version:
 
-```powershell
 $bugReport = &{
-    $p = Get-Module -Name Pester -ListAvailable | Select-Object -First 1
+    $p = Get-Command Invoke-Pester | Select-Object -ExpandProperty Module
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE.md:31]: This will work only if Pester is imported. The original solution will show what would be automatically imported. This is trade off between succeeding in when this is run in a clean session, and being more accurate. I would use the current solution. Or we can move the script to a gist and consider both options. 

Provide the result of this script:
```powershell
(Invoke-WebRequest -Uri "https://git.io/JTinj" -UseBasicParsing).Content | Invoke-Expression
```
This will run this gist: https://gist.github.com/nohwnd/3c78fb0e452dd4a49a8810c155023e33

Or run it manually:

```powershell
$bugReport = & {
    $i = Get-Command Invoke-Pester -ErrorAction SilentlyContinue | Select-Object -ExpandProperty Module
    $m = if ($i) { $i } else { Get-Module -Name Pester -ListAvailable | Select-Object -First 1 }
    $psv = $PSVersionTable.PSVersion
    $pre = $m.PrivateData -and $m.PrivateData.PSData -and $m.PrivateData.PSData.PreRelease
    $pv = if ($pre) { "$($m.Version)-$($m.PrivateData.PSData.PreRelease)" } else { $m.Version }
    $pi = if ($i) { " (available)" } else { " (imported)"} 
    "Pester version     : " + $pv + " " + $m.Path + $pi
    "PowerShell version : " + $psv
    "OS version         : " + [System.Environment]::OSVersion.VersionString
}
$bugReport
$bugReport | clip
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+---
+name: 🐛 Bug Report
+about: Report erros and unexpected behavior
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/bug_report.md:3]:
```suggestion
about: Report errors or unexpected behavior
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+title: 'My descriptive bug title'
+labels: ''
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/bug_report.md:5]:
```suggestion
labels: 'bug'
```

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -1,80 +1,38 @@
-## 1. General summary of the issue
-
 <!--
 
-Please provide a descriptive title of the issue in the field 'Title' too.
+Please provide a descriptive title in the field 'Title' too.
 
-Please be aware that Pester version 3.4.0 - which is shipped with Windows 10 and Windows Server 2016 - is not supported anymore.
+Please be aware that Pester version 3.4.0 - which is shipped with Windows 10 and Windows Server 2016 - is not supported anymore. Please update Pester to the latest version. See [Installation and update guide](https://pester.dev/docs/introduction/installation).
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE.md:5]: I would rephrase this to something like Try updating to the latest version of Pester. Only Pester 4.10.x and 5.x.x are supported, try updating to the latest version to see if that solves your problem.

The current phrasing lists a lot of details, that most people won't care about.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+assignees: ''
+
+---
+
+<!--
+
+Thank you for using Pester and taking the time to report this issue!
+
+Please be aware that Pester version 3.4.0 - which is shipped with Windows 10 and Windows Server 2016 - is not supported anymore.
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]: This can be removed. Updating is mentioned below.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+- Please update Pester and retest your code before submitting a bug report. See [Installation and update guide](https://pester.dev/docs/introduction/installation).
+- Search for existing issues.
+- Pester 5 introduced breaking changes and some features were removed or are not yet migrated. See [Breaking changes](https://github.com/pester/Pester#breaking-changes)
+
+-->
+
+## General summary of the issue
+
+
+## Describe your environment
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/bug_report.md:25]: Link to the same script as above. Probably just the invoke-webrequest and link to gist is enough, so we don't have to manually update it in multiple places.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+```
+-->
+
+## Expected Behavior
+
+<!-- A clear and concise description of what you expected to happen. -->
+
+## Current Behavior
+
+<!-- What happens instead of the expected behavior.. -->
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/bug_report.md:60]: nit: use just one dot

File: .github/ISSUE_TEMPLATE/commandhelp_issue.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+labels: 'Documentation'
+assignees: ''
+
+---
+
+<!--
+
+Thank you for using Pester and taking the time to report this issue!
+
+Known issue: Documentation is mostly out of date for all commands. See [Breaking changes](https://github.com/pester/Pester#breaking-changes)
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/commandhelp_issue.md:14]:
```suggestion
Known issue: Documentation of Pester 5 is mostly out of date for all commands. See [Breaking changes](https://github.com/pester/Pester#breaking-changes)
```

File: .github/ISSUE_TEMPLATE/commandhelp_issue.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+Command Reference documentation on the website are generated from comment-based help in the functions and should be reported here.
+
+All other documentation issues on https://pester.dev should be reported in the docs-repo.
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/commandhelp_issue.md:18]: link to docs repo?

File: .github/ISSUE_TEMPLATE/config.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+    - name: Need help?
+      url: https://pester.dev/docs/contributing/introduction#where-to-get-support
+      about: Check out the recommended communities available for usage questions before creating a general issue
```
Comment (nohwnd) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
      about: Join our online communities to get help with "How to do _____ with Pester?" questions.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -1,23 +1,36 @@
...
+If your pull request resolves a reported issue, please mention it by using `Fix #<issue_number>` on a new line, this will close the linked issue automatically when this PR is merged. For more info see: [Closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/).
 
 If your pull request integrates Pester with another system, please tell us how the change can be tested.
 
-Please remember to update [the Pester wiki](https://github.com/pester/Pester/wiki) if needed.
+-->
+
+## PR Checklist
+
+- [ ] PR has meaning title
```
Comment (nohwnd) [inline .github/PULL_REQUEST_TEMPLATE.md:21]:
```suggestion
- [ ] PR has meaningful title
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -1,23 +1,36 @@
...
+- [ ] Summary describes changes
+- [ ] PR is ready to be merge
```
Comment (nohwnd) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
- [ ] PR is ready to be merged
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 13]
```diff
@@ -1,23 +1,36 @@
...
+    - Run `./build.ps1 -Clean; ./test.ps1 -NoBuild`. Use  a new PowerShell process when C# code is changed.
+- [ ] Tests are added/update *(if required)*
+- [ ] Documentation is updated/added *(if required)*
 
-Before you continue, please review [Contributing to Pester](https://github.com/pester/Pester/wiki/Contributing-to-Pester) and [Development rules - technical](https://github.com/pester/Pester/wiki/Developement-rules---technical).
+<!--
+
+Before you continue, please review [Contributing to Pester](https://pester.dev/docs/contributing/introduction).
 
 Our continuous integration system doesn't send any notifications about failed tests. Please return to the opened pull request (after ~60 minutes) to check if is everything OK.
```
Comment (nohwnd) [inline .github/PULL_REQUEST_TEMPLATE.md:34]:
```suggestion
Our continuous integration system doesn't send any notifications about failed tests. Please return to the opened pull request (after ~60 minutes) to check if everything is OK.
```

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (fflaten) [reply .github/ISSUE_TEMPLATE.md:31]: `Get-Command Invoke-Pester` will autoload the same version you get from the failsafe if it's not already loaded. At the same time it works with manually imported module ex. from bin folder in git-repo. Or am I missing something?

I really like that gist-solution though! Do we think users (who don't open the link to read) would find it scary? I know similar is used for installation-guides in other Powershell modules, vscode ++, so might be good.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (fflaten) [reply .github/ISSUE_TEMPLATE/bug_report.md:14]: Full stop on v4 bugs reports too? "Please update Pester" was referring to latest v4 or v5, however v3 is done - won't be fixed.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (fflaten) [reply .github/ISSUE_TEMPLATE.md:31]: Btw. last line needs to be removed. "clip" is windows-only and "Set-Clipboard" is PSv5+

Comment (nohwnd) [reply .github/ISSUE_TEMPLATE.md]: Updated the gist. Removed clip and reverted to your approach. The shortlink seems to be pointing to a cached version that is in the middle of the changes, hopefully this will go away automatically.
